### PR TITLE
[ci:component:github.com/gardener/etcd-druid:v0.7.3->v0.8.0]

### PR DESCRIPTION
**Pull request by ghost**
**Release Notes**:
``` bugfix operator github.com/gardener/etcd-druid #312 @stoyanr
The default CPU and memory limits for `etcd` and `backup-restore` containers have been removed to enable removal of limits via the `Etcd` resource.
```

``` bugfix operator github.com/gardener/etcd-druid #310 @rfranzke
A bug has been fixed which caused the `etcd-druid` not removing its finalizers from referenced secrets in `Etcd` resources when those references changed.
```

``` noteworthy developer github.com/gardener/etcd-druid #309 @rfranzke
The CA data key is now configurable under `.spec.etcd.tls.tlsCASecretRef.dataKey`. It still defaults to `ca.crt` if not provided.
```

``` bugfix operator github.com/gardener/etcd-druid #306 @ishan16696
A bug has been fix which caused the Nil pointer exception in EtcdCopyBackupsTask.
```

``` improvement operator github.com/gardener/etcd-backup-restore #445 @timuthy
The used Golang version was updated to `1.17`.
```

``` improvement operator github.com/gardener/etcd-backup-restore #440 @sibucan
Fixed etcd-backup-restore exiting immediately on SIGTERM without proper context cancellation
```

``` feature operator github.com/gardener/etcd-backup-restore #435 @ishan16696
To Dynamically load Iaas credentials, added support to pass the credentials through file mount.
```

``` feature operator github.com/gardener/etcd-backup-restore #435 @ishan16696
The support to pass the credentials through environment variables will get deprecated by `etcd-backup-restore` v0.18.0
```

``` improvement operator github.com/gardener/etcd-backup-restore #431 @sibucan
* Added support for non-AWS S3-compatible providers by specifying a custom endpoint.
```

``` improvement operator github.com/gardener/etcd-backup-restore #411 @ishan16696
Failed backups won't fail the readinessProbe thus etcd is allowed to keep serving the incoming traffic.
```

``` improvement operator github.com/gardener/etcd-backup-restore #411 @ishan16696
Added CLI flags `--backoff-multiplier`, `--backoff-attempt-limit` and `--backoff-threshold-time` to configure the exponential-backoff mechanism.
```

``` improvement operator github.com/gardener/etcd-backup-restore #405 @ishan16696
Upgrade the Go from v1.14 to v1.16
```

``` feature operator github.com/gardener/etcd-backup-restore #353 @ishan16696
Added the leader-election to the backup-restore. Added support to allow only backup-restore Leader to take/upload the snapshots and trigger the defragmentation.
```

``` feature operator github.com/gardener/etcd-druid #301 @ishan16696
To Dynamically load Iaas credentials, added support to pass the credentials through secret mount.
```

``` feature operator github.com/gardener/etcd-druid #301 @ishan16696
Set File Path through Env: `<ProviderName>_APPLICATION_CREDENTIALS`
```

``` other operator github.com/gardener/etcd-druid #300 @kris94
The ETCD backup restore is now configured to support Local provider in container environment.
```

``` other operator github.com/gardener/etcd-druid #294 @timuthy
The Golang version which is used to build Etcd-Druid was updated to `1.17.6`.
```

``` other operator github.com/gardener/etcd-druid #293 @timuthy
Please be aware that Etcd-Druid needs to re-create the etcd `StatefulSet` if the etcd cluster is scaled up from `1 -> x` for the first time.
```

``` feature operator github.com/gardener/etcd-druid #285 @ishan16696
Leader-election specifications can be configured through Etcd resource spec via `.spec.backup.leaderElection`.
```

``` breaking operator github.com/gardener/etcd-druid #281 @acumino
The default leader election resource lock of `etcd-druid` has been changed from `configmapsleases` to `leases`.
Please make sure, that you had at least `etcd-druid@v0.5.0 ` running before upgrading so that it has successfully acquired leadership with the hybrid resource lock (`configmapsleases`) at least once.
```

``` feature developer github.com/gardener/etcd-druid #277 @rfranzke
When `--disable-etcd-serviceaccount-automount` is set to `true` then the `.automountServiceAccountToken` will be set to `false` for the `ServiceAccount` created for etcd.
```

``` other operator github.com/gardener/etcd-druid #276 @timuthy
The amount of active reconciliations has been limited to improve performance, CPU, memory and network consumption.
```

``` feature operator github.com/gardener/etcd-druid #273 @timuthy
A new service (`<etcd-name>-peer`) for etcd peer communication (default port `2380`) is now created by Etcd-Druid.
```

``` breaking operator github.com/gardener/etcd-druid #273 @timuthy
Using the etcd `client` service for server communication (default port `2380`) has been deprecated. The port will be removed from the service in the near future. If necessary, switch to the new `peer` service instead.
```

``` breaking operator github.com/gardener/etcd-druid #273 @timuthy
The claiming logic for services has been removed from Etcd-Druid. This means that existing `service` objects cannot be adopted anymore but a new and dedicated object is created. Please check any usages for already adopted `services` and switch to the dedicated `<etcd-name>-client` service.
```

``` other operator github.com/gardener/etcd-druid #270 @aaronfern
Updated labels used in compaction job to differentiate them from etcd pods. This allows for pod scheduling policies to schedule compaction jobs on predetermined nodes
```

``` bugfix operator github.com/gardener/etcd-druid #263 @timuthy
A bug has been fixed that led to multiple update conflicts when the `etcd` resource was reconciled.
```

``` feature operator github.com/gardener/etcd-druid #262 @timuthy
Etcd-Druid now creates member `Lease` objects which enables the heartbeat functionality for etcd members. Along the way a new flag `--etcd-member-unknown-threshold` was introduced. It determines the duration after which a etcd member's state is considered `unknown` when the member `Lease` is not renewed.
```

``` breaking developer github.com/gardener/etcd-druid #261 @timuthy
Make target `install-requirements` was dropped. Instead, required 3rd party binaries are automatically installed to a local bin dir (./hack/tools/bin).
```

``` improvement developer github.com/gardener/etcd-druid #261 @timuthy
The handling for 3rd party binaries (e.g. `controller-gen`), usually required for Make targets, has been improved. Instead of installing those tools to a global directory, a dedicated local directory of in the code repository is used (`hack/tools/bin`).
```

``` bugfix operator github.com/gardener/etcd-druid #260 @timuthy
A bug has been fixed which caused the `etcd.status.clusterSize` only being set for new `etcd` resources
```

``` other operator github.com/gardener/etcd-druid #250 @breuerfelix
feature operator
Deploying and maintaining the correct PodDisruptionBudget configuration according to Etcd resource status
```

```noteworthy operator
The `etcd` pods for shoot control planes will be restarted during the first shoot reconciliation.
```

### CODE REVIEW

**Comment by ishan16696** [comment]
what about also adding a release note as warning of:  ⚠️ `restart of etcd.`

**Comment by rfranzke** [comment]
@ishan16696 done

File: pkg/operation/botanist/etcd.go [hunk 1]
```diff
@@ -108,12 +109,18 @@ func (b *Botanist) DeployEtcd(ctx context.Context) error {
 			return err
 		}
 
+		var backupLeaderElection *config.ETCDBackupLeaderElection
+		if b.Config != nil && b.Config.ETCDConfig != nil {
+			backupLeaderElection = b.Config.ETCDConfig.BackupLeaderElection
```
Comment (ishan16696) [inline pkg/operation/botanist/etcd.go:114]: what if `ETCDConfig` don't have `BackupLeaderElection` set ?

**Review by ishan16696 — COMMENTED** [review]
Thanks @rfranzke for PR,
Overall looks good, just 1 doubt

File: pkg/operation/botanist/etcd.go (hunk 1 above)
Comment (rfranzke) [reply pkg/operation/botanist/etcd.go:114]: Then `backupLeaderElection` is still `nil` like before in L112, nothing happens.

Comment (ishan16696) [reply pkg/operation/botanist/etcd.go:114]: ok, then at Line no 123 we are setting `LeaderElection` to nil. Will this be an issue?

Comment (stoyanr) [reply pkg/operation/botanist/etcd.go:114]: Then `backupLeaderElection` will be `nil`, which should still be safe, or?

Comment (rfranzke) [reply pkg/operation/botanist/etcd.go:114]: No, this is no issue and equivalent to not having any configuration for the backup leader election. In this case, `etcd-druid` defaults to whatever its default values are.

Comment (ishan16696) [reply pkg/operation/botanist/etcd.go:114]: ok, now I got it. It won’t be an issue.
thanks

**Review by ishan16696 — APPROVED** [review]
LGTM!

**Review by rfranzke — APPROVED** [review]

**Comment by rfranzke** [comment]
Sure, thanks @stoyanr!

**Comment by stoyanr** [comment]
I reconciled an existing cluster with this PR successfully, `etcd-druid` is now on the new release and I see the `--disable-etcd-serviceaccount-automount=true`. However, I don't see the projected volume injected in etcd pods, could this be something on my side?

**Comment by stoyanr** [comment]
I also added the `backupLeaderElection` settings and saw them correctly propagated to the `etcd-main` Etcd resource and from there to the `etcd-main` pods.

**Comment by rfranzke** [comment]
Well, it seems that the `etcd-druid` does not reconcile the `ServiceAccount` as expected (despite the fact that it happily logs "Reconciling serviceaccount"). The respective function only creates the object and reconciles the labels, nothing else: https://github.com/gardener/etcd-druid/blob/ce1007cd58b58b201dd643b7bf8f47e59cf412b0/controllers/etcd_controller.go#L817-L857.

For now, I have removed the reference to #4659 and will follow up on this outside of this PR. It seems we should release v0.8.1 with a fix.

**Comment by rfranzke** [comment]
I opened https://github.com/gardener/etcd-druid/issues/315

**Comment by rfranzke** [comment]
@stoyanr Is there anything else from your side?

**Comment by stoyanr** [comment]
I also tested creating a new shoot with this PR. There, I do see the projected volume injected in etcd pods. So it seems to work for new shoots, but not for existing shoots.

> @stoyanr Is there anything else from your side?

No, everything else looks good!

/lgtm

**Review by stoyanr — APPROVED** [review]

**Pull request merged**